enterprise/cypher/compiled-expressions/src/test/scala/org/neo4j/cypher/internal/runtime/compiled/expressions/CodeGenerationTest.scala

```diff
@@ -0,0 +1,383 @@
+    compile(function("acos", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+
+  test("cos function") {
+    val arg = random.nextDouble()
+    compile(function("cos", literalFloat(arg))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(Math.cos(arg)))
+    compile(function("cos", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+
+  test("cot function") {
+    val arg = random.nextDouble()
+    compile(function("cot", literalFloat(arg))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(1 / Math.tan(arg)))
+    compile(function("cot", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+
+  test("atan function") {
+    val arg = random.nextDouble()
+    compile(function("atan", literalFloat(arg))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(Math.atan(arg)))
+    compile(function("atan", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+
+  test("tan function") {
+    val arg = random.nextDouble()
+    compile(function("tan", literalFloat(arg))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(Math.tan(arg)))
+    compile(function("tan", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+
+  test("ceil function") {
+    val arg = random.nextDouble()
+    compile(function("ceil", literalFloat(arg))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(Math.ceil(arg)))
+    compile(function("ceil", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+
+  test("floor function") {
+    val arg = random.nextDouble()
+    compile(function("floor", literalFloat(arg))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(Math.floor(arg)))
+    compile(function("floor", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+
+  test("abs function") {
+    compile(function("abs", literalFloat(3.2))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(3.2))
+    compile(function("abs", literalFloat(-3.2))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(3.2))
+    compile(function("abs", literalInt(3))).compute(ctx, tx, EMPTY_MAP) should equal(longValue(3))
+    compile(function("abs", literalInt(-3))).compute(ctx, tx, EMPTY_MAP) should equal(longValue(3))
+    compile(function("abs", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(Values.NO_VALUE)
+  }
+
+  test("add numbers") {
+    // Given
+    val expression = add(literalInt(42), literalInt(10))
+
+    // When
+    val compiled = compile(expression)
+
+    // Then
+    compiled.compute(ctx, tx, EMPTY_MAP) should equal(longValue(52))
+  }
+
+  test("add with NO_VALUE") {
+    // Given
+    val expression = add(parameter("a"), parameter("b"))
+
+    // When
+    val compiled = compile(expression)
+
+    // Then
+    compiled.compute(ctx, tx, map(Array("a", "b"), Array(longValue(42), NO_VALUE))) should equal(NO_VALUE)
+    compiled.compute(ctx, tx, map(Array("a", "b"), Array(NO_VALUE, longValue(42)))) should equal(NO_VALUE)
+  }
+
+  test("add strings") {
```

Comment:
There was a billion overloaded add code paths. I guess there are a few more to test?